plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2027,6 +2027,7 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
 
         // Table.snapshot method returns null if there is no matching snapshot
         Snapshot snapshot = requireNonNull(icebergTable.snapshot(optimizeHandle.snapshotId().get()), "snapshot is null");
+        rewriteFiles.dataSequenceNumber(snapshot.sequenceNumber());
```

Comment:
Let's add a code comment here that this is needed to avoid commit conflicts between OPTIMIZE and addition of equality deletes by a concurrent writer